pkg/reconciler/revision/resources/deploy.go

```diff
@@ -170,6 +176,24 @@ func makePodSpec(rev *v1alpha1.Revision, loggingConfig *logging.Config, tracingC
 	return podSpec, nil
 }
 
+func verifySecrets(rev *v1alpha1.Revision, secretLister v1.SecretLister) error {
+	var errStr string
+	for _, volume := range rev.Spec.Volumes {
+		if secret := volume.VolumeSource.Secret; secret != nil {
+			if _, err := secretLister.Secrets(rev.Namespace).Get(secret.SecretName); err != nil {
+				if errStr != "" {
+					errStr += "\n"
```

Comment:
Yeah I agree it would be better if these can be api field errors, but I don't think we can return an api field error here. I also looked at potentially having the secrets validation as part of revision validation, but we won't have access to the secret lister.

Reply:
The field errors just implement the error interface though. In fact, they do roughly the same string building you're doing here as well but providing a little more info.